src/terminal/adapter/adaptDispatchGraphics.cpp

```diff
@@ -262,7 +279,15 @@ void AdaptDispatch::_ApplyGraphicsOptions(const VTParameters options,
 {
     for (size_t i = 0; i < options.size();)
     {
-        i += _ApplyGraphicsOption(options, i, attr);
+        if (options.hasSubParamsFor(i))
+        {
+            _ApplyGraphicsOptionSubParam(options.at(i), options.subParamsFor(i), attr);
+            i += 1;
+        }
+        else
+        {
+            i += _ApplyGraphicsOption(options, i, attr);
+        }

```

Comment:
I just realised that this sub parameter test is in the wrong place. It needs to be a level down inside the `_ApplyGraphicsOption` method, because that gets called directly by the `DECRARA` implementation. As it is now, `DECRARA` won't get the sub parameter support.

Reply:
done.